### PR TITLE
PAYARA-2443 - update Jackson

### PR DESCRIPTION
**Pull request by mikecroft**
Fixes #2327 

Builds OK locally.

### CODE REVIEW

**Comment by mikecroft** [comment]
Payara 5 doesn't *need* to be updated to fix the linked issue, but here I'm updating Jackson to the latest stable release so it might make sense to update Payara 5 too.

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
Jenkins, quick test JDK7 please

File: nucleus/pom.xml [hunk 1]
```diff
@@ -190,7 +190,7 @@
         <gmbal.version>4.0.0-b001</gmbal.version>
         <antlr.version>2.7.7</antlr.version>
         <jersey.version>2.25.1</jersey.version>
-        <jackson.version>2.8.8</jackson.version>
+        <jackson.version>2.9.3</jackson.version>
```
Comment (Cousjava) [inline nucleus/pom.xml]: 2.9.4 is the latest patch

**Comment by Pandrex247** [comment]
We need to be careful about not breaking JDK7 builds - do we know if these new versions are JDK8 only?
The Jenkins error from the JDK7 run is unrelated - something JDK8 only has been merged in to master.

File: nucleus/pom.xml (hunk 1 above)
Comment (mikecroft) [reply nucleus/pom.xml]: So it is!
https://github.com/FasterXML/jackson-databind/releases

**Comment by mikecroft** [comment]
@Pandrex247 the fail is here:
https://jenkins.payara.fish/job/PayaraQuickBuild-JDK7/111/console

Complaining about this line which has been there for months:
https://github.com/payara/Payara/blame/master/appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java#L1439

**Comment by mikecroft** [comment]
@Pandrex247 

> #### JDK
> No changes from 2.8: Java 7 features available for all modules (meaning, need to compile on Java 7) except:
> 
> * jackson-annotations and jackson-core (streaming) still Java 6 only
> * Java 8 modules, Kotlin module require Java 8 for building, runtime
> * Other modules SHOULD run on Java 6 with reduced features (i.e. avoid loading Java 7 types)

https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.9#jdk

**Comment by mikecroft** [comment]
jenkins test please

**Pull request merged**